source/preprocessor.tex

```diff
@@ -662,104 +666,39 @@
 \indextext{header unit!preprocessing}%
 \indextext{macro import|(}%
 
-\begin{bnf}
-\nontermdef{import-seq}\br
-    \opt{top-level-token-seq} \opt{\terminal{export}} \terminal{import}
-\end{bnf}
-
-\begin{bnf}
-\nontermdef{top-level-token-seq}\br
-    \descr{any \grammarterm{pp-balanced-token-seq} ending in \terminal{;} or \terminal{\}}}
-\end{bnf}
-
 \begin{bnf}
 \nontermdef{pp-import}\br
-    \terminal{import} header-name \opt{pp-import-suffix} \terminal{;}\br
-    \terminal{import} header-name-tokens \opt{pp-import-suffix} \terminal{;}
-\end{bnf}
-
-\begin{bnf}
-\nontermdef{pp-import-suffix}\br
-    pp-import-suffix-token\br
-    pp-import-suffix pp-import-suffix-token
-\end{bnf}
-
-\begin{bnf}
-\nontermdef{pp-import-suffix-token}\br
-    \descr{any \grammarterm{pp-balanced-token} other than \terminal{;}}
-\end{bnf}
-
-\begin{bnf}
-\nontermdef{pp-balanced-token-seq}\br
-    pp-balanced-token\br
-    pp-balanced-token-seq pp-balanced-token
-\end{bnf}
-
-\begin{bnf}
-\nontermdef{pp-balanced-token}\br
-    pp-ldelim \opt{pp-balanced-token-seq} pp-rdelim\br
-    \descr{any \grammarterm{preprocessing-token} other than a \grammarterm{pp-ldelim} or \grammarterm{pp-rdelim}}
-\end{bnf}
-
-\begin{bnf}
-\nontermdef{pp-ldelim} \descr{one of}\br
-  \terminal{(    [    \{    <:    <\%}
-\end{bnf}
-
-\begin{bnf}
-\nontermdef{pp-rdelim} \descr{one of}\br
-  \terminal{)    ]    \}    :>    \%>}
+    \opt{\terminal{export}} \terminal{import} header-name \opt{pp-tokens} \terminal{;} new-line\br
+    \opt{\terminal{export}} \terminal{import} header-name-tokens \opt{pp-tokens} \terminal{;} new-line\br
+    \opt{\terminal{export}} \terminal{import} pp-tokens \terminal{;} new-line
 \end{bnf}
 
 \pnum
-A sequence of \grammarterm{preprocessing-token}{s}
-matching the form of a \grammarterm{pp-import}
+The preprocessing tokens after the \tcode{import} preprocessing token
+in the \tcode{import} \grammarterm{control-line}
+are processed just as in normal text
+(i.e., each identifier currently defined as a macro name
+is replaced by its replacement list of preprocessing tokens).
+An \tcode{import} directive
+matching the first two forms of a \grammarterm{pp-import}
 instructs the preprocessor to import macros
 from the header unit\iref{module.import}
 denoted by the \grammarterm{header-name}.
-A \grammarterm{pp-import} is only recognized
-when the sequence of tokens
-produced by phase 4 of translation
-up to the \tcode{import} token
-forms an \grammarterm{import-seq},
-and the \tcode{import} token is not within
-the \grammarterm{header-name-tokens} or \grammarterm{pp-import-suffix}
-of another \grammarterm{pp-import}.
-The \tcode{;} \grammarterm{preprocessing-token}
-terminating a \grammarterm{pp-import}
-shall not have been produced by
-macro replacement\iref{cpp.replace}.
-The \defnadj{point of}{macro import} for a \grammarterm{pp-import} is
-immediately after the \tcode{;} terminating the \grammarterm{pp-import}.
+The \defnadj{point of}{macro import} for the
```

Comment:
I know this was in the original text, but I think this definition should be changed to: `\defnadj{point of import}{macro}`.
Likewise, while it's outside the scope of this paper, the \defnx{}s under "macro definition" should also be moved under "macro", i.e.: 
`\defnx{point of definition}{macro definition!point of definition}`
should become:
`\defnadj{point of definition}{macro}`

Reply:
We can only use \defnadj{x}{y} for stuff that should read "x y" in the resulting document, but I get your point.

I've changed stuff.

Reply:
Everything "macro" is now under "macro".
Everything "point of "is now under "point of" with cross-references.

Reply:
> We can only use \defnadj{x}{y} for stuff that should read "x y" ...
Duh, yeah :).  Glad you got my point.  Thanks for the fixes.